### PR TITLE
update outdated dependencies; fixes #123

### LINKED ISSUE #123: Outdated dependencies
```
$ npm outdated

Package        Current  Wanted    Latest  Location
github           0.2.3   0.2.3     0.2.4  github
moment           2.9.0   2.9.0    2.10.3  moment
nodemailer       1.3.0   1.3.0     1.3.4  nodemailer
underscore       1.7.0   1.7.0     1.8.3  underscore
promise          6.1.0   6.1.0     7.0.3  promise
hubot           2.11.0  2.11.0    2.13.2  hubot
request         2.51.0  2.51.0    2.58.0  request
hubot-cron       0.2.7   0.2.7     0.2.9  hubot-cron
googlemaps      0.1.20  0.1.20     1.0.1  googlemaps
hubot-scripts   2.5.16  2.5.16    2.16.1  hubot-scripts
hubot-hipchat    2.7.5   2.7.5  2.12.0-5  hubot-hipchat
hipchatter       0.1.3   0.1.3     0.2.0  hipchatter
```

### PR DESCRIPTION
**Pull request by macklinu**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,19 +22,17 @@
   "dependencies": {
     "cool-ascii-faces": "1.3.3",
     "datejs": "1.0.0-rc3",
-    "github": "0.2.3",
-    "googlemaps": "0.1.20",
```
Comment (macklinu) [inline package.json]: unused from what I can tell

File: package.json [hunk 2]
```diff
@@ -22,19 +22,17 @@
...
-    "hipchatter": "~0.1",
```
Comment (macklinu) [inline package.json]: unused

File: package.json [hunk 3]
```diff
@@ -22,19 +22,17 @@
...
-    "hubot": "2.11.0",
-    "hubot-cron": "0.2.7",
-    "hubot-hipchat": "2.7.5",
-    "hubot-scripts": "2.5.16",
-    "moment": "2.9.0",
+    "github": "0.2.4",
+    "hubot": "2.13.2",
+    "hubot-cron": "0.2.9",
```
Comment (macklinu) [inline package.json]: is this used anywhere?

File: package.json [hunk 4]
```diff
@@ -22,19 +22,17 @@
...
+    "hubot-hipchat": "2.12.0-5",
+    "hubot-scripts": "2.16.1",
+    "moment": "2.10.3",
     "mqtt": "^1.0.10",
-    "nodemailer": "1.3.0",
-    "promise": "6.1.0",
-    "request": "~2.51",
-    "underscore": "1.7.0"
+    "nodemailer": "1.3.4",
+    "promise": "7.0.3",
```
Comment (macklinu) [inline package.json]: used in [`github-project-status.coffee`](https://github.com/detroit-labs/hubot-badger/blob/master/scripts/github-project-status.coffee)

File: package.json [hunk 5]
```diff
@@ -22,19 +22,17 @@
...
-    "hubot-cron": "0.2.7",
-    "hubot-hipchat": "2.7.5",
-    "hubot-scripts": "2.5.16",
-    "moment": "2.9.0",
+    "github": "0.2.4",
+    "hubot": "2.13.2",
+    "hubot-cron": "0.2.9",
+    "hubot-hipchat": "2.12.0-5",
+    "hubot-scripts": "2.16.1",
+    "moment": "2.10.3",
```
Comment (macklinu) [inline package.json]: used in [`coffeepot.coffee`](https://github.com/detroit-labs/hubot-badger/blob/master/scripts/coffeepot.coffee)

**Comment by nwest** [comment]
```sh
nwest@ender:~DL/hubot-badger/scripts(masterλ)
» ack require
announcements-email.coffee
23:nodemailer = require 'nodemailer'

badger.coffee
22:crypto = require('crypto')

broadside.coffee
15:request = require "request"

coffeepot.coffee
16:moment = require("moment")

dlreddit.coffee
15:request = require "request"

events.coffee
10:util = require 'util'

face.coffee
16:cool = require 'cool-ascii-faces'

github-project-status.coffee
23:_ = require 'underscore'
24:githubAPI = require 'github'
26:Promise = require 'promise'

httpd.coffee
20:spawn = require('child_process').spawn

ostrta.coffee
19:_ = require 'underscore'

redlight.coffee
15:mqtt = require("mqtt")

roles.coffee
47:_ = require 'underscore'

storage.coffee
9:Util = require "util"

tigers.coffee
16:require 'datejs'
17:_ = require 'underscore'
```

**Comment by macklinu** [comment]
Way better

On Tue, Jun 23, 2015, 9:22 AM Nate West <notifications@github.com> wrote:

> nwest@ender:~DL/hubot-badger/scripts(masterλ)
> » ack require
> announcements-email.coffee
> 23:nodemailer = require 'nodemailer'
>
> badger.coffee
> 22:crypto = require('crypto')
>
> broadside.coffee
> 15:request = require "request"
>
> coffeepot.coffee
> 16:moment = require("moment")
>
> dlreddit.coffee
> 15:request = require "request"
>
> events.coffee
> 10:util = require 'util'
>
> face.coffee
> 16:cool = require 'cool-ascii-faces'
>
> github-project-status.coffee
> 23:_ = require 'underscore'
> 24:githubAPI = require 'github'
> 26:Promise = require 'promise'
>
> httpd.coffee
> 20:spawn = require('child_process').spawn
>
> ostrta.coffee
> 19:_ = require 'underscore'
>
> redlight.coffee
> 15:mqtt = require("mqtt")
>
> roles.coffee
> 47:_ = require 'underscore'
>
> storage.coffee
> 9:Util = require "util"
>
> tigers.coffee
> 16:require 'datejs'
> 17:_ = require 'underscore'
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/detroit-labs/hubot-badger/pull/124#issuecomment-114499013>
> .
>
-- 

Thanks,
Macklin Underdown
Detroit Labs

**Comment by nwest** [comment]
This doesn't take into account the dependencies used in hubot-scripts.json from [here](https://github.com/github/hubot-scripts).

**Comment by macklinu** [comment]
@nwest taking the `hubot-scripts.json` dependencies into consideration, I piped each file in there into ack and determined all of those scripts' dependencies:

```
Url   = require "url"
Redis = require "redis"
qs = require "querystring"
url = require "url"
crypto = require "crypto"
http = require "http"
querystring = require 'querystring'
```

**Comment by macklinu** [comment]
I tested the following scripts successfully with these dependency updates:

```
coffeepot.coffee
github-project-status.coffee
roles.coffee
tigers.coffee
```

Haven't tried the rest of our custom scripts that rely on updated dependencies.

**Comment by macklinu** [comment]
This update looks like it may have broken [`ostrta.coffee`](https://github.com/detroit-labs/hubot-badger/blob/master/scripts/ostrta.coffee) - I can't get the script to respond to anything locally. @nwest @maschall can you confirm this?

**Pull request merged**